### PR TITLE
get data only when online

### PR DESCRIPTION
**Pull request by JeremyDezalos**

### CODE REVIEW

File: app/src/main/java/com/github/wnder/GuessPreviewActivity.java [hunk 1]
```diff
@@ -92,16 +102,23 @@ protected void onStart() {
...
     private void openGuessActivity() {
-        Intent intent = new Intent(this, GuessLocationActivity.class);
+        if (networkInfo.isNetworkAvailable()){
+            Intent intent = new Intent(this, GuessLocationActivity.class);
 
         intent.putExtra(GuessLocationActivity.EXTRA_CAMERA_LAT, user.getPositionFromGPS((LocationManager) getSystemService(Context.LOCATION_SERVICE), this).getLatitude());
         intent.putExtra(GuessLocationActivity.EXTRA_CAMERA_LNG, user.getPositionFromGPS((LocationManager) getSystemService(Context.LOCATION_SERVICE), this).getLongitude());
         intent.putExtra(GuessLocationActivity.EXTRA_PICTURE_LAT, pictureLat);
         intent.putExtra(GuessLocationActivity.EXTRA_PICTURE_LNG, pictureLng);
         intent.putExtra(GuessLocationActivity.EXTRA_PICTURE_ID, pictureID);
```
Comment (rkuenzi-epfl) [inline app/src/main/java/com/github/wnder/GuessPreviewActivity.java:112]: Indetation issue.

File: app/src/main/java/com/github/wnder/networkService/NetworkModule.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import dagger.Provides;
+import dagger.hilt.InstallIn;
+import dagger.hilt.android.qualifiers.ApplicationContext;
+import dagger.hilt.components.SingletonComponent;
+
+@Module
+@InstallIn(SingletonComponent.class)
+public abstract class NetworkModule {
+    @Binds
+    public abstract NetworkService bindPicturesDatabase(NetworkInformation networkInformationImpl);
```
Comment (rkuenzi-epfl) [inline app/src/main/java/com/github/wnder/networkService/NetworkModule.java:20]: Copy paste error here.

File: app/src/main/java/com/github/wnder/networkService/NetworkService.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package com.github.wnder.networkService;
+
+public interface NetworkService {
+    boolean isNetworkAvailable();
```
Comment (rkuenzi-epfl) [inline app/src/main/java/com/github/wnder/networkService/NetworkService.java:4]: No javadoc.

File: app/src/main/java/com/github/wnder/networkService/NetworkInformation.java [hunk 4]
```diff
@@ -1,24 +1,27 @@
...
+    @Inject
+    public NetworkInformation(ConnectivityManager manager){
+        this.manager = manager;
     }
 
     /**
      * Checks if the cellular or wifi connection is available
-     * @param ctx Context of the application
      * @return true if we are connected to internet
      */
```
Comment (rkuenzi-epfl) [inline app/src/main/java/com/github/wnder/networkService/NetworkInformation.java:22]: It should be moved to NetworkService I think

File: app/src/main/java/com/github/wnder/AlertBuilder.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import android.content.DialogInterface;
+
+import androidx.appcompat.app.AlertDialog;
+
+public final class AlertBuilder {
+    private AlertBuilder(){
+        //Non instanciable class
+    }
+
+    public static AlertDialog createAlert(String title, String body, Context ctx){
```
Comment (rkuenzi-epfl) [inline app/src/main/java/com/github/wnder/AlertBuilder.java:14]: As the alert is always the same (for now at least) could we just create a "noConnectionAlert(Context ctx)"? (That would reuse this method)

**Review by rkuenzi-epfl — CHANGES_REQUESTED** [review]
Please correct at least the copy/paste error. The tests when there's alerts looks quite complete which is nice.

File: app/src/main/java/com/github/wnder/AlertBuilder.java (hunk 5 above)
Comment (JeremyDezalos) [reply app/src/main/java/com/github/wnder/AlertBuilder.java:14]: The body changes so it is not possible to do that

Comment (rkuenzi-epfl) [reply app/src/main/java/com/github/wnder/AlertBuilder.java:14]: I though it was the same everywhere mb.

**Review by rkuenzi-epfl — APPROVED** [review]

File: app/src/main/java/com/github/wnder/TakePictureActivity.java [hunk 6]
```diff
@@ -64,14 +76,20 @@ protected void onCreate(Bundle savedInstanceState) {
...
+
+                Intent intent = new Intent(this, UploadActivity.class);
+                intent.putExtra(HAS_SUCCEEDED, hasSucceeded);
+                startActivity(intent);
+                this.finish();
+            }
+            else{
+                AlertDialog alert = AlertBuilder.noConnectionAlert(getString(R.string.no_connection), getString(R.string.no_internet_body), this);
+                alert.show();
+            }
```
Comment (BoostedBoat) [inline app/src/main/java/com/github/wnder/TakePictureActivity.java:92]: Maybe we could let the user store the picture in his gallery even without connection, and only prevent him from uploading it to the db

**Review by BoostedBoat — APPROVED** [review]
For the code climate issues, you could make one more method taking a class as argument (or anything else) that would choose to call on or the other activity depending on the argument passed, don't know if it's necessary though.

**Pull request merged**